### PR TITLE
feat: add Root Solver

### PR DESCRIPTION
**Pull request by vishwa2710**

### CODE REVIEW

File: include/OpenSpaceToolkit/Astrodynamics/RootSolver.hpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+namespace ostk
+{
+namespace astro
+{
+
+using ostk::core::types::Size;
+using ostk::core::types::Real;
+
+/// @brief                      A Root Solver is a tool for finding the roots of equations.
```
Comment (Derollez) [inline include/OpenSpaceToolkit/Astrodynamics/RootSolver.hpp:19]:
```suggestion
/// @brief                      A Root Solver is a method for finding the roots (a.k.a. zeros) of an equation.
```

File: include/OpenSpaceToolkit/Astrodynamics/RootSolver.hpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+class RootSolver
+{
+   public:
+    struct Solution
```
Comment (Derollez) [inline include/OpenSpaceToolkit/Astrodynamics/RootSolver.hpp:23]: Nice, similarly, I think we can have a RootSolver Settings (or Config) Struct (not necessarily now but later if you had more methods) to store all the settings and parameters for the rootSolvers. They usually have a bunch in common (maxIters, tols, etc)

File: src/OpenSpaceToolkit/Astrodynamics/RootSolver.cpp [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+void RootSolver::print(std::ostream& anOutputStream, bool displayDecorator) const
+{
+    displayDecorator ? ostk::core::utils::Print::Header(anOutputStream, "Root Solver") : void();
+
+    ostk::core::utils::Print::Line(anOutputStream) << "Maximum Iterations Count:" << maximumIterationsCount_;
+    ostk::core::utils::Print::Line(anOutputStream) << "Number of Digits:" << numberOfDigits_;
+
+    displayDecorator ? ostk::core::utils::Print::Footer(anOutputStream) : void();
+}
```
Comment (Derollez) [inline src/OpenSpaceToolkit/Astrodynamics/RootSolver.cpp:38]: Move after getter implementation

Comment (Derollez) [reply src/OpenSpaceToolkit/Astrodynamics/RootSolver.cpp:38]: and after functions

File: include/OpenSpaceToolkit/Astrodynamics/RootSolver.hpp [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    /// @brief              Solve for root given a function, an initial guess and function direction
+    ///
+    /// @param              [in] aFunction A function
+    /// @param              [in] anInitialGuess An initial guess
+    /// @param              [in] isRising A boolean indicating whether the function is rising
+    ///
+    /// @return             The solution
+
+    Solution solve(
+        const std::function<double(const double&)>& aFunction, const double& anInitialGuess, const bool& isRising
```
Comment (Derollez) [inline include/OpenSpaceToolkit/Astrodynamics/RootSolver.hpp:80]: Just out of curiosity (let's not do that now), boost has some methods for multi-dimensional functions? see some good applications for multi-variate polynomials (laaaater)

**Review by Derollez — APPROVED** [review]
LGTM, thanks for making this a legit class !

File: include/OpenSpaceToolkit/Astrodynamics/RootSolver.hpp (hunk 4 above)
Comment (vishwa2710) [reply include/OpenSpaceToolkit/Astrodynamics/RootSolver.hpp:80]: Interestingggg. I will do some research.

File: include/OpenSpaceToolkit/Astrodynamics/RootSolver.hpp (hunk 2 above)
Comment (vishwa2710) [reply include/OpenSpaceToolkit/Astrodynamics/RootSolver.hpp:23]: Yeah for sure. I'm not sure how this class will grow so we can definitely do that in the future.

**Pull request merged**